nanogpt/CMakeLists.txt

```
@@ -0,0 +1,55 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# ### One-time setup ###
+#
+# Configure the CMake build system. It's good practice to do this whenever
+# cloning or pulling the upstream repo. Once this is done, you don't need to do
+# it again until you pull from the upstream repo again.
+#
+# NOTE: If your `buck2` binary is not on the PATH, you can change this line to
+# say something like `-DBUCK2=/tmp/buck2` to point directly to the tool.
+#[[
+  (rm -rf cmake-out \
+    && mkdir cmake-out \
+    && cd cmake-out \
+    && cmake -DBUCK2=buck2 ..)
+]]
+#
+# ### Build ###
+#
+# NOTE: The `-j` argument specifies how many jobs/processes to use when
+# building, and tends to speed up the build significantly. It's typical to use
+# "core count + 1" as the `-j` value.
+# ~~~
+# cmake --build cmake-out --target nanogpt_runner -j9
+# ~~~
+
+cmake_minimum_required(VERSION 3.19)
+project(nanogpt_runner)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+
+# Set options for executorch build.
+option(EXECUTORCH_BUILD_EXTENSION_DATA_LOADER "" ON)
+option(EXECUTORCH_BUILD_EXTENSION_MODULE "" ON)
+option(EXECUTORCH_BUILD_XNNPACK "" ON)
+
+# Include the executorch subdirectory.
+add_subdirectory(
+    ${PROJECT_SOURCE_DIR}/../executorch
+    ${CMAKE_BINARY_DIR}/executorch)
```

It's safer to use CMAKE_CURRENT_SOURCE_DIR to refer to the current directory. PROJECT_SOURCE_DIR can change if this project is included as a subdirectory. Though I think I might have been the one who used PROJECT_SOURCE_DIR originally :)

```suggestion
add_subdirectory(
    ${CMAKE_CURRENT_SOURCE_DIR}/../executorch
    ${CMAKE_BINARY_DIR}/executorch)
```